### ISSUE TITLE
Make torchani's training setup identical to neurochem 

### ISSUE THREAD

**Issue by farhadrgh** [aiqm/torchani#243]
After a thorough comparison between Neurochem and Torchani, I have found a few differences as explained briefly in the following:

- [x] NC skips the weight decay for the first layer of each atomic network, even if the weight decay parameter is provided in the input. This was the main source of difference in training performance.(#275)
- [x] In NC the weights are initialized to He Kaiming normal and the biases to zero. (#245, #275)
- [x] Loss in NC is scaled by sqrt(natoms). (#275)
- [x] NC uses AdamW optimizer to update the weights and SGD for the biases. (#275)
- [x] There were a couple of discrepancies between theory and the implementation of AdamW in Torchani (#261).
- [x] NC scale the gradients of atomic networks by the number of atoms in that batch, and compute the gradient from SumSquaredError instead of MeanSquaredError. (will be fixed in Neurochem)
- [ ] NC reshuffle the batches after every epoch (tracked in #248, no much improvement as tested, low priority)
